plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1047 @@
+            raise AnsibleError('URL is malformed: {0}'.format(to_native(url)))
+        return True
+
+    def _connect_to_socket(self):
+        """connect to lxd socket
+
+        Connect to lxd socket by provided url or defaults
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            AnsibleError
+        Returns:
+            None"""
+        error_storage = {}
+        url_list = [self.get_option('url'), self.SNAP_SOCKET_URL, self.SOCKET_URL]
+        urls = (url for url in url_list if self.validate_url(url))
+        for url in urls:
+            try:
+                socket_connection = LXDClient(url, self.client_key, self.client_cert, self.debug)
+                return socket_connection
+            except LXDClientException as err:
+                error_storage[url] = err
+        raise AnsibleError('No connection to the socket: {0}'.format(to_native(error_storage)))
+
+    def _get_networks(self):
+        """Get Networknames
+
+        Returns all network config names
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            list(names): names of all network_configs"""
+        network_configs = self.socket.do('GET', '/1.0/networks')
+        # e.g. {'type': 'sync',
+        #       'status': 'Success',
+        #       'status_code': 200,
+        #       'operation': '',
+        #       'error_code': 0,
+        #       'error': '',
+        #       'metadata': ['/1.0/networks/lxdbr0']}
+        names = []
+        for index in network_configs['metadata']:
+            # e.g. ['', '1.0', 'networks', 'lxdbr0']
+            names.append(index.split('/')[3])
+        return names
+
+    def _get_containers(self):
+        """Get Containernames
+
+        Returns all containernames
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            list(names): names of all containers"""
+        # e.g. {'type': 'sync',
+        #       'status': 'Success',
+        #       'status_code': 200,
+        #       'operation': '',
+        #       'error_code': 0,
+        #       'error': '',
+        #       'metadata': ['/1.0/containers/udemy-ansible-ubuntu-2004']}
+        containers = self.socket.do('GET', '/1.0/containers')
+        names = []
+        for index in containers['metadata']:
+            # e.g. ['', '1.0', 'containers', 'udemy-ansible-ubuntu-2004']
+            names.append(index.split('/')[3])
+        return names
+
+    def _get_config(self, branch, name):
+        """Get inventory of container
+
+        Get config of container
+
+        Args:
+            str(branch): Name oft the API-Branch
+            str(name): Name of Container
+        Kwargs:
+            None
+        Source:
+            https://github.com/lxc/lxd/blob/master/doc/rest-api.md
+        Raises:
+            None
```

Comment:
That's very java-ish of you :-) (been there, done that), but given the lack of declared exceptions in Python, I think it is virtually impossible to do this accurately. In fact, in line 374 you are using the fact that this method may raise a `LXDClientException`, and I don't blame you for missing that. Personally, I'd rather not to promise what I cannot deliver, so I would remove `Raises` from the comment. I understand there might be some controversy there and will not press the point, it is a matter of personal preferences.

Reply:
I took that from https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html only to use sphinx out of the box. Have Ansible any doctext style guide?

Reply:
If it has one I haven't gone that far yet 😁
That said, I would in general question whether doing things one way or another actually improves matters or not.

Reply:
At this point I am absolutely unemotional and like to join a style guide. It is currently important for me that other systems, as mentioned Sphinx, understand the Sysntax and that I don't have to bend everything. When I do the latter, I end up with a pipline that doesn't work "out of the box" and only works for me.
If you and Felix say "out with it", it's a 5 minute thing: O)

Reply:
Different styles are in use in Ansible, so there's none that's the best. I'd personally leave the `None` sections away, since they tend to make the docstring unnecessarily large, but as I said that's personal preference.